src/Sarif/Core/StackFrame.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Reflection;
+using System.Text;
+using DotNetStackFrame = System.Diagnostics.StackFrame;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// A stack frame of a SARIF stack.
+    /// </summary>
+    public partial class StackFrame
+    {
+        internal const string IN = " in ";
+        internal const string AT = "   at ";
+        internal const string LINE = ":line";
+
+        /// <summary>
+        /// Creates a SARIF StackFrame instance from a .NET StackFrame instance
+        /// </summary>
+        /// <param name="stackTrace"></param>
+        /// <returns></returns>
+        public static StackFrame Create(System.Diagnostics.StackFrame dotNetStackFrame)
+        {
+            // This value is -1 if not present
+            string fileName = dotNetStackFrame.GetFileName();
+            int ilOffset = dotNetStackFrame.GetILOffset();
+            int nativeOffset = dotNetStackFrame.GetNativeOffset();
+            MethodBase methodBase = dotNetStackFrame.GetMethod();
+            Assembly assembly = methodBase?.DeclaringType.Assembly;
+            string fullyQualifiedName = CreateFullyQualifiedName(methodBase);
+
+            PhysicalLocation physicalLocation;
+            
+            if (fileName != null)
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Uri = new Uri(fileName),
+                    Region = new Region()
+                    {
+                        StartLine = dotNetStackFrame.GetFileLineNumber(),
+                        StartColumn = dotNetStackFrame.GetFileColumnNumber(),
+                        Offset = ilOffset
```

Comment:
Here we are again :-) We decided to get rid of `byteOffset` and `charOffset`, and replace them with just `Offset`, which -- depending on whether the `Region` is a "text region" (as determined by whether `StartLine` is > 0) -- is interpreted either as characters or bytes. So in this case, `Offset` would be interpreted as a character offset.

I'm more than happy to resurrect `charOffset` and `byteOffset`. This is the first scenario I've seen where a byte offset coexists with line and column information.

From the spec:

> When a region object represents a binary region, it shall contain a property named offset which shall have an integer value equal to the byte offset from the beginning of the file of the first byte in the region.

> When a region object represents a text region, the offset property may be present. In this case, it represents the character offset from the beginning of the file of the first character in the region